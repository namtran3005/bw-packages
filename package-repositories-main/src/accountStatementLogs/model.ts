import mongooseLeanId from 'mongoose-lean-id';
import { Document, Schema } from "mongoose";
import { mainConnection } from '../mainConnection';

export interface AccountStatementLogsDoc extends Document {
  id: string;
  userId: string;
  locale: string;
  firstName: string;
  solarisPersonSolarisId: string;
  solarisAccountSolarisId: string;
  email: string;
  sent: boolean;
  reason: string;
  createdAt: Date;
  updatedAt: Date;
}

export const accountStatementLogsSchema: Schema = new Schema(
  {
    userId: {
      type: String,
      required: true,
      index: true,
      unique: true
    },
    locale: {
      type: String,
    },
    firstName: {
      type: String,
    },
    solarisPersonSolarisId: {
      type: String,
      index: true,
    },
    solarisAccountSolarisId: {
      type: String,
      index: true,
    },
    email: {
      type: String,
    },
    sent: {
      type: Boolean,
    },
    reason: {
      type: String,
    }
  },
  { 
    timestamps: true,
    collection: "account-statement-logs",
  }
);

accountStatementLogsSchema.plugin(mongooseLeanId);

export const AccountStatementLogsModel = mainConnection.db.model<AccountStatementLogsDoc>(
  'AccountStatementLogs',
  accountStatementLogsSchema
);
