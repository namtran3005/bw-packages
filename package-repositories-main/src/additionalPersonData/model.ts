import mongooseLeanId from 'mongoose-lean-id';
import {
  additionalPersonDataSchema,
  AdditionalPersonDataDoc,
} from '@bitwala-cryptobank-squad/package-schemas';

import { mainConnection } from '../mainConnection';

additionalPersonDataSchema.plugin(mongooseLeanId);

export const AdditionalPersonDataModel = mainConnection.db.model<AdditionalPersonDataDoc>(
  'AdditionalDataPerson',
  additionalPersonDataSchema
);
