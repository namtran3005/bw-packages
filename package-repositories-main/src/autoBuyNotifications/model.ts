import mongooseLeanId from 'mongoose-lean-id';
import {
  autoBuyNotificationsSchema,
  AutoBuyNotificationsDoc,
} from '@bitwala-cryptobank-squad/package-schemas';
import { mainConnection } from '../mainConnection';

autoBuyNotificationsSchema.plugin(mongooseLeanId);

export const AutoBuyNotificationsModel = mainConnection.db.model<
  AutoBuyNotificationsDoc
>('AutoBuyNotifications', autoBuyNotificationsSchema);
