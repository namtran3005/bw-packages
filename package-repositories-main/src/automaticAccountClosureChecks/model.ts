import {
  AutomaticAccountClosureChecksDoc,
  automaticAccountClosureChecksSchema,
} from '@bitwala-cryptobank-squad/package-schemas';
import { mainConnection } from '../mainConnection';

export const AutomaticAccountClosureChecksModel = mainConnection.db.model<
AutomaticAccountClosureChecksDoc
>('AutomaticAccountClosureChecks', automaticAccountClosureChecksSchema);
