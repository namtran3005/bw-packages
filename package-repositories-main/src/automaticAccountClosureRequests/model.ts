import {
  AutomaticAccountClosureRequestsDoc,
  automaticAccountClosureRequestsSchema,
} from '@bitwala-cryptobank-squad/package-schemas';
import { mainConnection } from '../mainConnection';

export const AutomaticAccountClosureRequestsModel = mainConnection.db.model<
AutomaticAccountClosureRequestsDoc
>('AutomaticAccountClosureRequests', automaticAccountClosureRequestsSchema);
