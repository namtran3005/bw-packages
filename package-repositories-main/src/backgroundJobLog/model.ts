import mongooseLeanId from 'mongoose-lean-id';
import {
  backgroundJobLogsSchema,
  BackgroundJobLogDoc,
} from '@bitwala-cryptobank-squad/package-schemas';
import { mainConnection } from '../mainConnection';

backgroundJobLogsSchema.plugin(mongooseLeanId);

export const BackgroundJobLogModel = mainConnection.db.model<BackgroundJobLogDoc>(
  'BackgroundJobLog',
  backgroundJobLogsSchema
);
