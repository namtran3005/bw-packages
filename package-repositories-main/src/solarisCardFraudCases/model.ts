import mongooseLeanId from 'mongoose-lean-id';
import {
  solarisCardFraudCaseSchema,
  SolarisCardFraudCaseDoc,
} from '@bitwala-cryptobank-squad/package-schemas';
import { mainConnection } from '../mainConnection';

solarisCardFraudCaseSchema.plugin(mongooseLeanId);

export const SolarisCardFraudCaseModel = mainConnection.db.model<
  SolarisCardFraudCaseDoc
>('SolarisCardFraudCases', solarisCardFraudCaseSchema);
