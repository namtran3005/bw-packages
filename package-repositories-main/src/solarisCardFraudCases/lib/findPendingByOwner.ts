import { SolarisCardFraudCaseDoc } from '@bitwala-cryptobank-squad/package-schemas';
import {
  CardFraudCaseResolution,
  CardFraudCaseTransactionStatus
} from '@bitwala-cryptobank-squad/package-solaris';
import { DocumentDefinition } from 'mongoose';
import { SolarisCardFraudCaseModel } from '../model';

export const findPendingByOwner = (
  owner: string
): Promise<DocumentDefinition<SolarisCardFraudCaseDoc>[]> => {
  return SolarisCardFraudCaseModel.find({
    owner,
    resolution: CardFraudCaseResolution.PENDING,
    respondUntil: {
      $gt: new Date(),
    },
    'cardTransaction.status': CardFraudCaseTransactionStatus.DECLINED,
  })
    .sort({ respondUntil: -1 })
    .lean()
    .exec();
};
