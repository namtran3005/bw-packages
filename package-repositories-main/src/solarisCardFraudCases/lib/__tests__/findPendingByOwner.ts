/* eslint-disable @typescript-eslint/no-explicit-any */
import { SolarisCardFraudCaseModel } from '../../model';

import { findPendingByOwner } from '../findPendingByOwner';

const mockFindVal: any = {
  sort: jest.fn().mockImplementation(() => mockFindVal),
  lean: jest.fn().mockImplementation(() => mockFindVal),
  exec: jest.fn().mockImplementation(() => Promise.resolve('EXEC')),
};
const mockDate = new Date('2018-05-05');

describe('findPendingByOwner (find card fraud case docs by status)', () => {
  beforeAll(() => {
    jest
      .spyOn(SolarisCardFraudCaseModel, 'find')
      .mockImplementation(() => mockFindVal);
    jest.spyOn(global, 'Date').mockImplementation(() => mockDate as any);
  });
  beforeEach(() => {
    jest.clearAllMocks();
  });
  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should find sorted with lean and return a promise', async () => {
    const res = await findPendingByOwner('ownerId');
    expect(SolarisCardFraudCaseModel.find).toMatchSnapshot();
    expect(mockFindVal.sort).toBeCalledWith({ respondUntil: -1 });
    expect(mockFindVal.lean).toBeCalled();
    expect(mockFindVal.exec).toBeCalled();
    expect(res).toBe('EXEC');
  });
});
