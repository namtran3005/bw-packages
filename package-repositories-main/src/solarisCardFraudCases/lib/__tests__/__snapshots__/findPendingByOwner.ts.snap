// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findPendingByOwner (find card fraud case docs by status) should find sorted with lean and return a promise 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "cardTransaction.status": "DECLINED",
        "owner": "ownerId",
        "resolution": "PENDING",
        "respondUntil": Object {
          "$gt": 2018-05-05T00:00:00.000Z,
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Object {
        "exec": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": Promise {},
            },
          ],
        },
        "lean": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": [Circular],
            },
          ],
        },
        "sort": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "respondUntil": -1,
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": [Circular],
            },
          ],
        },
      },
    },
  ],
}
`;
